From a6f46fda177e12a7ee9b86c24d17226013a8d4ae Mon Sep 17 00:00:00 2001
From: Thiago Macieira <thiago.macieira@intel.com>
Date: Tue, 14 Jun 2016 15:14:04 -0700
Subject: [PATCH] Fix build on PPC: PPC is defined to 1

 $ powerpc-poky-linux-gcc -dM -E -xc /dev/null | grep -i PPC
 #define _ARCH_PPC 1
 #define __PPC__ 1
 #define __PPC 1
 #define PPC 1

Not on PPC64, though (only __PPC__ and __PPC64__ are defined).

Change-Id: Ib57b52598e2f452985e9fffd145812f5098e441d
---
 src/plugins/qmakeprojectmanager/makefileparse.cpp | 4 ++--
 src/plugins/qmakeprojectmanager/qmakestep.cpp     | 8 ++++----
 src/plugins/qmakeprojectmanager/qmakestep.h       | 2 +-
 tests/manual/shootout/tst_codesize.cpp            | 2 +-
 4 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/src/plugins/qmakeprojectmanager/makefileparse.cpp b/src/plugins/qmakeprojectmanager/makefileparse.cpp
index a52f22c..8b63e87 100644
--- a/src/plugins/qmakeprojectmanager/makefileparse.cpp
+++ b/src/plugins/qmakeprojectmanager/makefileparse.cpp
@@ -167,12 +167,12 @@ void MakeFileParse::parseAssignments(QList<QMakeAssignment> *assignments)
                         m_config.archConfig = QMakeStepConfig::NoArch;
                 } else if (value == QLatin1String("ppc")) {
                     if (qa.op == QLatin1String("+="))
-                        m_config.archConfig = QMakeStepConfig::PPC;
+                        m_config.archConfig = QMakeStepConfig::PowerPC;
                     else
                         m_config.archConfig = QMakeStepConfig::NoArch;
                 } else if (value == QLatin1String("ppc64")) {
                     if (qa.op == QLatin1String("+="))
-                        m_config.archConfig = QMakeStepConfig::PPC64;
+                        m_config.archConfig = QMakeStepConfig::PowerPC64;
                     else
                         m_config.archConfig = QMakeStepConfig::NoArch;
                 } else if (value == QLatin1String("iphonesimulator")) {
diff --git a/src/plugins/qmakeprojectmanager/qmakestep.cpp b/src/plugins/qmakeprojectmanager/qmakestep.cpp
index 8c00e09..33d64ed 100644
--- a/src/plugins/qmakeprojectmanager/qmakestep.cpp
+++ b/src/plugins/qmakeprojectmanager/qmakestep.cpp
@@ -806,9 +806,9 @@ QMakeStepConfig::TargetArchConfig QMakeStepConfig::targetArchFor(const Abi &targ
                 arch = QMakeStepConfig::X86_64;
         } else if (targetAbi.architecture() == ProjectExplorer::Abi::PowerPCArchitecture) {
             if (targetAbi.wordWidth() == 32)
-                arch = QMakeStepConfig::PPC;
+                arch = QMakeStepConfig::PowerPC;
             else if (targetAbi.wordWidth() == 64)
-                arch = QMakeStepConfig::PPC64;
+                arch = QMakeStepConfig::PowerPC64;
         }
     }
     return arch;
@@ -838,9 +838,9 @@ QStringList QMakeStepConfig::toArguments() const
         arguments << QLatin1String("CONFIG+=x86");
     else if (archConfig == X86_64)
         arguments << QLatin1String("CONFIG+=x86_64");
-    else if (archConfig == PPC)
+    else if (archConfig == PowerPC)
         arguments << QLatin1String("CONFIG+=ppc");
-    else if (archConfig == PPC64)
+    else if (archConfig == PowerPC64)
         arguments << QLatin1String("CONFIG+=ppc64");
 
     // TODO: make that depend on the actual Qt version that is used
diff --git a/src/plugins/qmakeprojectmanager/qmakestep.h b/src/plugins/qmakeprojectmanager/qmakestep.h
index 3509e85..9a048d4 100644
--- a/src/plugins/qmakeprojectmanager/qmakestep.h
+++ b/src/plugins/qmakeprojectmanager/qmakestep.h
@@ -76,7 +76,7 @@ class QMAKEPROJECTMANAGER_EXPORT QMakeStepConfig
 {
 public:
     enum TargetArchConfig {
-        NoArch, X86, X86_64, PPC, PPC64
+        NoArch, X86, X86_64, PowerPC, PowerPC64
     };
 
     enum OsType {
diff --git a/tests/manual/shootout/tst_codesize.cpp b/tests/manual/shootout/tst_codesize.cpp
index 02762d8..6300539 100644
--- a/tests/manual/shootout/tst_codesize.cpp
+++ b/tests/manual/shootout/tst_codesize.cpp
@@ -360,7 +360,7 @@ void tst_CodeSize::codesize_data()
     QByteArray std_tie_code =
         "struct QMakeStepConfig\n"
         "{\n"
-        "    enum TargetArchConfig { NoArch, X86, X86_64, PPC, PPC64 };\n"
+        "    enum TargetArchConfig { NoArch, X86, X86_64, PowerPC, PowerPC64 };\n"
         "    enum OsType { NoOsType, IphoneSimulator, IphoneOS };\n"
         "\n"
         "    QMakeStepConfig()\n"
-- 
2.8.1

